### PR TITLE
Add `LogicalPoint`, `PhysicalPoint` and `Point`

### PR DESCRIPTION
**Pull request by amrbashir**
part of #2395

- [x] Tested on all platforms changed
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [x] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

**Comment by daxpedda** [comment]
Looking at #2395 I don't really follow what the motivation here is exactly.
Is there some previous related discussion?

**Comment by daxpedda** [comment]
My question is specifically how does `Logical/PhysicalPoint` relate to that? Why can that not be done with the types we have now?

**Comment by daxpedda** [comment]
Ah, I see, thank you!

**Review by kchibisov — CHANGES_REQUESTED** [review]
Should probably update the docs in `### Position and Size types` with the point stuff.

**Comment by kchibisov** [comment]
Besides.

@amrbashir given that you had an interest in such crate in the first place, how you deal with e.g. rounding? For example in winit we have special rounding rules on Wayland, because it's rounding is a bit special, but maybe we should define such things in this crate as well, so the `to_logical/to_physical` will account for rounding and make the default like it was before?

**Comment by amrbashir** [comment]
We didn't give it much thought, since our windows and macOS backend were forked from `winit` and the current behavior of `to_logical/to_physical` have been working fine for us.

**Comment by amrbashir** [comment]
I just realized that the naming `Point` might be confusing when used with width/height as it conveys position than a dimension, maybe it should be renamed to `LogicalUnit`, `PhysicalUnit` and `Unit` or `PixelUnit` ?

**Comment by kchibisov** [comment]
>I just realized that the naming Point might be confusing when used with width/height as it conveys position than a dimension, maybe it should be renamed to LogicalUnit, PhysicalUnit and Unit or PixelUnit ?

Maybe `LogicalUnit`/`PhysicalUnit` is better.

**Review by kchibisov — APPROVED** [review]

File: dpi/CHANGELOG.md
Comment (daxpedda) [inline dpi/CHANGELOG.md:1]: I guess we are fine with not also moving this changelog to docs.rs like in #3496?

**Review by daxpedda — APPROVED** [review]

File: dpi/CHANGELOG.md
Comment (kchibisov) [reply dpi/CHANGELOG.md:1]: Yeah, it's not that important.

**Review by notgull — APPROVED** [review]

**Pull request merged**